### PR TITLE
test: allow to run tests in parallel

### PR DESCRIPTION
**Pull request by robertsLando**

### CODE REVIEW

**Comment by robertsLando** [comment]
After many tries seems that this is not possible at all and doesn't have any real benefit and there are still problems:

- PNPM doesn't support concurrent installs
- MACOS codesign utility doesn't allow concurrent signs commands

Another way could be to run `npm run test:host && npm run test:18 && npm run test:20` in different steps

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        type: string
+        required: true
+        default: 'install'
+jobs:
+  test:
+    strategy:
+      fail-fast: false # prevent test to stop if one fails
+      matrix:
+        node-version: [18.x, 20.x]
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (q0rban) [inline .github/workflows/test.yml:17]: what if we also expanded the matrix based on the individual tests? That way, each test would run in it's own runner asynchronously, and would also make it easier to see which test failed.

Comment (robertsLando) [reply .github/workflows/test.yml:17]: That will not be beneficial as there will be too runners spawn and github has a limit on runners, you will end up having lot runners waiting

Comment (q0rban) [reply .github/workflows/test.yml:17]: Good to know! What is the limit? We've not run into that limit [here](https://github.com/TugboatQA/images/actions/runs/11300906163). Say the limit is 50, even if we hit the limit and had to wait, wouldn't that be faster?

Comment (robertsLando) [reply .github/workflows/test.yml:17]: I see the limit in macos runner images, I think they give you at most 2/3 macos runner for time

**Pull request merged**